src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs

```diff
@@ -745,7 +745,7 @@ public MethodSymbol ReduceExtensionMethod(TypeSymbol receiverType, CSharpCompila
                 throw new ArgumentNullException(nameof(receiverType));
             }
 
-            if (!this.IsExtensionMethod || this.MethodKind == MethodKind.ReducedExtension)
+            if (!this.IsExtensionMethod || this.MethodKind == MethodKind.ReducedExtension || receiverType.IsVoidType())
```

Comment:
>|| receiverType.IsVoidType() [](http://example.com/codeflow?start=90&length=28)

Do we need to make similar change for VB?

Reply:
I don't think so. VB has the check [here](https://sourceroslyn.io/#Microsoft.CodeAnalysis.VisualBasic/Binding/Binder_Lookup.vb,1928), which also guards extension method look-up.